site/content/arangodb/4.0/graphs/enterprisegraphs/_index.md

```
@@ -25,21 +25,22 @@ from a `general-graph` to an `enterprise-graph`.
-whenever possible. This approach provides significant advantages as it
-minimizes the impact of having suboptimal sharding keys defined when creating
-the graph.
-
-This means that, when using EnterpriseGraphs, the `smartGraphAttribute` is
-**not** required. As a consequence, you **cannot** define `_key` values on
-edges. 
+EnterpriseGraphs use hash-based sharding similar to General Graphs that
+pseudo-randomly distribute nodes across shards, but EnterpriseGraphs ensure that
+the adjacent edges of nodes are co-located in the same shard. As a consequence,
+you **cannot** define `_key` values for edges. You can also **not modify** the
+`_from ` and `_to` edge attributes because they contain the sharding key which
```

markdownlint-cli2 (0.22.1)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)